### ISSUE TITLE
--no-hmr reloads the app for every single file updated (even if all updated at same time)

### ISSUE THREAD

**Issue by SephReed** [NativeScript/NativeScript#7680]
**Describe the bug**
I have sass converting scss to css files.  If I modify a file, it will update like so:
> Compiled src/components/common/RangesValue/rangesValue.scss to app/components/common/RangesValue/rangesValue.css.
> Compiled src/app.android.scss to app/app.android.css.
> Compiled src/app.ios.scss to app/app.ios.css.

For each of these three files, the app will restart, three times in a row. 

**To Reproduce**
Run tns debug ios|android --no-hmr.  Do anything that updates multiple files at the same time.

**Expected behavior**
It should only update once.  There should be some form of batching.  ie:

```js
// batch updates together by 200ms burndown timer
let timeout;
wepback.onFileChange(() => {
  if (timeout) { clearTimeout(timeout); }
  timeout = setTimeout(() => {
    timeout = undefined;
    updateAndRestartApp();
  }, 200);
});
```

**Comment by SephReed** [comment]
I suppose I should mention, as a result of this issue, the forced webpack flow, and the fact that hmr does not work for nativescript-core projects, I'm currently unable to work on my project at all without watching 5-10s of repeated restarts for every single css update.  Which is to say, it's a dead project until this gets fixed.

**Comment by tsonevn** [comment]
Hi @SephReed 
I tested the described scenario in a newly created sample project with latest {N} on both(`iOS` and `Android`), however, was unable to reproduce a similar problem while using `--no-hmr` flag. For your convenience, I am attaching the project that I used for testing. You can check whether you will be able to recreate the problem with it. On that matter, can you provide some more info about your environment(`CLI`, `tns-core-modules`, `tns-android`, `tns-ios`, `nativescript-angular` versions)?
Also, it will help if you can send us a sample project that can be used for recreating the issue.

**Comment by SephReed** [comment]
I ended up restructuring my project yesterday and it's working now.

I had been running a js-core project, where I had typescript and sass that I was personally transpiling from a src folder to an app folder.  Every transpiled file was triggering this.

Now that everything is webpack, I don't have to worry about having js files mixed in with my ts files, so I was able to switch over to a true ts-core project structure.

I still think a batching timer would be good to add.  I imagine if you create a script to overwrite multiple files in quick succession, you'll see what I'm talking about.
